backend/routers/filament_stock.py

```
@@ -0,0 +1,79 @@
+    filament_id: int,
+    filament: FilamentCreate,
+    db: Session = Depends(get_db)
+):
+    try:
+        updated = filament_service.update_filament(filament_id, filament, db)
+        if updated is None:
+            raise HTTPException(
+                status_code=404, detail="Filamento não encontrado")
+        return updated
+    except Exception as e:
+        raise HTTPException(status_code=500, detail=str(e))
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)